src/modules/accounting/extra_data.py

```diff
@@ -0,0 +1,153 @@
+import itertools
+from dataclasses import dataclass
+from enum import Enum
+
+from hexbytes import HexBytes
+
+from src.web3py.extensions.lido_validators import NodeOperatorGlobalIndex
+from src.web3py.typings import Web3
+
+
+class ItemType(Enum):
+    EXTRA_DATA_TYPE_STUCK_VALIDATORS = 1
+    EXTRA_DATA_TYPE_EXITED_VALIDATORS = 2
+
+
+class FormatList(Enum):
+    EXTRA_DATA_FORMAT_LIST_EMPTY = 1
+    # TODO old contracts have 0, new contracts have 1, don't forget to change it
+    EXTRA_DATA_FORMAT_LIST_NON_EMPTY = 1
+
+
+@dataclass
+class ItemPayload:
+    module_id: bytes
+    node_ops_count: bytes
+    node_operator_ids: bytes
+    vals_counts: bytes
+
+
+@dataclass
+class ExtraDataItem:
+    item_index: bytes
+    item_type: ItemType
+    item_payload: ItemPayload
+
+
+@dataclass
+class ExtraData:
+    extra_data: bytes
+    data_hash: HexBytes
+    format: int
+    items_count: int
+
+
+class ExtraDataService:
+    # Extra data is an array of items, each item being encoded as follows:
+    # |  3 bytes  | 2 bytes  |   X bytes   |
+    # | itemIndex | itemType | itemPayload |
+    #
+    # itemPayload format:
+    #  | 3 bytes  |   8 bytes    |  nodeOpsCount * 8 bytes  |  nodeOpsCount * 16 bytes  |
+    #  | moduleId | nodeOpsCount |      nodeOperatorIds     |   stuckOrExitedValsCount  |
+    class Lengths:
+        ITEM_INDEX = 3
+        ITEM_TYPE = 2
+        MODULE_ID = 3
+        NODE_OPS_COUNT = 8
+        NODE_OPERATOR_IDS = 8
+        STUCK_OR_EXITED_VALS_COUNT = 16
+
+    def collect(
+        self,
+        stuck_validators: dict[NodeOperatorGlobalIndex, int],
+        exited_validators: dict[NodeOperatorGlobalIndex, int],
+        max_items_in_payload_count: int,
+        max_items_count: int,
+    ) -> ExtraData:
+        stuck_payloads, rest_items = self.build_validators_payloads(stuck_validators, max_items_in_payload_count, max_items_count)
```

Comment:
To be consistent with a new version, please check out the following:

https://github.com/lidofinance/lido-dao/blob/e57517730c3e11a41e9cbc32ce018726722335b7/contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol#L54..L60

Reply:
Fixed